packages/mgt/src/components/mgt-planner/mgt-planner.ts

```diff
@@ -0,0 +1,845 @@
+ * @cssprop --task-new-add-button-disabled-background - {Color} Task new add button disabled background
+ * @cssprop --task-new-cancel-button-color - {Color} Task new cancel button color
+ * @cssprop --task-complete-background - {Color} Task complete background
+ * @cssprop --task-complete-border - {String} Task complete border
+ * @cssprop --task-complete-header-color - {Color} Task complete header color
+ * @cssprop --task-complete-detail-color - {Color} Task complete detail color
+ * @cssprop --task-complete-detail-icon-color - {Color} Task complete detail icon color
+ * @cssprop --tasks-background-color - {Color} Task background color
+ * @cssprop --task-icon-alignment - {String} Task icon alignment
+ * @cssprop --task-icon-background - {Color} Task icon color
+ * @cssprop --task-icon-background-completed - {Color} Task icon background color when completed
+ * @cssprop --task-icon-border - {String} Task icon border styles
+ * @cssprop --task-icon-border-completed - {String} Task icon border style when task is completed
+ * @cssprop --task-icon-border-radius - {String} Task icon border radius
+ * @cssprop --task-icon-color - {Color} Task icon color
+ * @cssprop --task-icon-color-completed - {Color} Task icon color when completed
+ */
+@customElement('mgt-planner')
+export class MgtPlanner extends MgtTasksBase {
+  /**
+   * Array of styles to apply to the element. The styles should be defined
+   * using the `css` tag function.
+   */
+  public static get styles() {
+    return styles;
+  }
+
+  /**
+   * if set, the component will only show tasks from this bucket or bucket
+   * @type {string}
+   */
+  @property({ attribute: 'target-bucket-id', type: String })
+  public targetBucketId: string;
+
+  /**
+   * if set, the component will first show tasks from this bucket or bucket
+   *
+   * @type {string}
+   * @memberof MgtTasks
+   */
+  @property({ attribute: 'initial-bucket-id', type: String })
+  public initialBucketId: string;
+
+  /**
+   * allows developer to define specific plan id
+   */
+  @property({ attribute: 'plan-id', type: String })
+  public planId: string;
+
+  @property() private _newTaskPlanId: string;
+  @property() private _newTaskBucketId: string;
+  @property() private _plans: GraphTypes.PlannerPlan[];
+  @property() private _buckets: GraphTypes.PlannerBucket[];
+  @property() private _tasks: GraphTypes.PlannerTask[];
+  @property() private _currentPlan: GraphTypes.PlannerPlan;
+  @property() private _currentBucket: GraphTypes.PlannerBucket;
+
+  private _isLoadingTasks: boolean;
+  private _loadingTasks: string[];
+  private _newTaskDueDate: Date;
+  private _graph: IGraph;
+  private _me: GraphTypes.User;
+
+  constructor() {
+    super();
+    this._newTaskPlanId = '';
+    this._newTaskBucketId = '';
+    this._plans = [];
+    this._buckets = [];
+    this._tasks = [];
+    this._currentPlan = null;
+    this._currentBucket = null;
+    this._loadingTasks = [];
```

Comment:
Need _isNewTaskVisible to determine whether or not to show the New tasks button. I took the following from mgt-tasks
``` 

  /**
   * Get whether new task view is visible
   *
   * @memberof MgtTasks
   */
  public get isNewTaskVisible() {
    return this._isNewTaskVisible;
  }

  /**
   * Set whether new task is visible
   *
   * @memberof MgtTasks
   */
  public set isNewTaskVisible(value: boolean) {
    this._isNewTaskVisible = value;
    if (!value) {
      this._newTaskDueDate = null;
      this._newTaskName = '';
      this._newTaskGroupId = '';
    }
  }
```